soroban-env-host/src/native_contract/account_contract.rs

```diff
@@ -84,11 +84,9 @@ pub(crate) fn check_account_contract_auth(
                 signature_args_vec.into(),
                 auth_context_vec.into(),
             ],
-            // Allow reentry for this function in order to do wallet admin ops
-            // within the auth framework. Maybe there is a more elegant way
-            // around this.
-            // TODO: check if there are security concerns about this.
-            true,
+            // Allow self reentry for this function in order to be able to do
+            // wallet admin ops using the auth framework itself.
+            ContractReentryMode::SelfAllowed,
```

Comment:
Just curious, do you have an example account contract somewhere that exercises re-entry?

Reply:
Sure, simple_account here does that and it is exercised using the built-in token contract.